lang/java/reef-io/src/main/java/org/apache/reef/io/watcher/WatcherConfiguration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.watcher;
+
+import org.apache.reef.driver.parameters.*;
+import org.apache.reef.io.watcher.param.Path;
+import org.apache.reef.tang.formats.*;
+import org.apache.reef.wake.time.Clock;
+
+/**
+ * ConfigurationModule for Watcher.
+ */
+public final class WatcherConfiguration extends ConfigurationModuleBuilder {
+
+  public static final OptionalImpl<EventStream> EVENT_STREAM = new OptionalImpl<>();
+
+  public static final OptionalParameter<String> PATH = new OptionalParameter<>();
+
+  public static final ConfigurationModule CONF = new WatcherConfiguration()
+      .bindImplementation(EventStream.class, EVENT_STREAM)
```

Comment:
I can imagine needing to have multiple `EventStreams`. Might be a good idea to use a `Set` here.